algoliasearch/src/main/java/com/algolia/search/clients/AnalyticsClient.java

```diff
@@ -0,0 +1,314 @@
+package com.algolia.search.clients;
+
+import com.algolia.search.exceptions.AlgoliaRuntimeException;
+import com.algolia.search.exceptions.LaunderThrowable;
+import com.algolia.search.http.AlgoliaHttpRequester;
+import com.algolia.search.http.IHttpRequester;
+import com.algolia.search.inputs.analytics.ABTest;
+import com.algolia.search.models.*;
+import com.algolia.search.objects.RequestOptions;
+import com.algolia.search.responses.ABTests;
+import com.algolia.search.transport.HttpTransport;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nonnull;
+
+@SuppressWarnings("WeakerAccess")
+public class AnalyticsClient {
+
+  private final HttpTransport transport;
+  private final AlgoliaConfig config;
+
+  public AnalyticsClient(@Nonnull String applicationID, @Nonnull String apiKey) {
+    this(new AnalyticsConfig(applicationID, apiKey));
+  }
+
+  public AnalyticsClient(@Nonnull AnalyticsConfig config) {
+    this(config, new AlgoliaHttpRequester(config));
+  }
+
+  public AnalyticsClient(@Nonnull AnalyticsConfig config, @Nonnull IHttpRequester httpRequester) {
+
+    Objects.requireNonNull(httpRequester, "An httpRequester is required.");
+    Objects.requireNonNull(config, "A configuration is required.");
```

Comment:
No needed as the parameters are already marked as @Nonnull

Reply:
@NonNull is only an annotation and won't throw an exception at runtime. I think it's better to have the double check to fail fast. IMO the combination of both is complementary.